lib/remap-lang/src/program.rs

```diff
@@ -89,8 +89,9 @@ impl Program {
         source: &str,
         function_definitions: &[Box<dyn Function>],
         constraint: Option<TypeConstraint>,
+        allow_regex_return: bool, // TODO: move this into a builder pattern
```

Comment:
I added this to make it easier to write some of our tests, that don't care about the return value.

Eventually, if/when VRL is its own project (it is, but it's still in the Vector repo right now), we'll change this so that Vector either handle regex return values, or we raise an error in Vector itself instead of here.

But the TL;DR is: please ignore this for now 😄